src/main/java/org/fever/filecreator/ClassArgumentFetcher.java

```diff
@@ -0,0 +1,96 @@
+package org.fever.filecreator;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.jetbrains.python.psi.*;
+import com.jetbrains.python.psi.search.PyClassInheritorsSearch;
+import com.jetbrains.python.psi.types.PyClassType;
+import com.jetbrains.python.psi.types.PyType;
+import com.jetbrains.python.psi.types.TypeEvalContext;
+import groovyjarjarantlr4.v4.misc.OrderedHashMap;
+import org.apache.commons.lang.ArrayUtils;
+import org.fever.fileresolver.DependencyInjectionFileResolverByClassName;
+import org.fever.utils.ClassArgumentParser;
+import org.fever.utils.IdentifierExtractor;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class ClassArgumentFetcher {
+    private static final int SELF_INDEX_IN_PARAMETER_LIST = 0;
+    public static Collection<IdentifierItem> getFqnOfInitArguments(PyClass originClass) {
+        List<IdentifierItem> identifiers = new ArrayList<>();
+        PyFile sourceCodeFile = (PyFile) originClass.getContainingFile();
+        Project project = sourceCodeFile.getProject();
+
+        TypeEvalContext context = TypeEvalContext.userInitiated(
+                project,
+                sourceCodeFile
+        );
+        Map<PyParameter, @Nullable PyClass> initParamsToClasses = getClassesFromInitParams(originClass, context);
+
+        for (Map.Entry<PyParameter, @Nullable PyClass> entry : initParamsToClasses.entrySet()) {
+            PyParameter parameter = entry.getKey();
+            PyClass parameterClass = entry.getValue();
+            String className = ClassArgumentParser.parse(parameter.getText());
+            List<String> implementationsClassNames = getAllImplementationsClassNames(parameterClass, className);
+            implementationsClassNames.add(className);
+
+            for (String implementationClassName : implementationsClassNames) {
+                PsiFile dependencyInjectionFile = DependencyInjectionFileResolverByClassName.resolve(project, implementationClassName);
+                String implementationIdentifier = IdentifierExtractor.extractIdentifierFromDIFile(dependencyInjectionFile);
+                IdentifierItem item = new IdentifierItem(implementationIdentifier, parameterClass, parameter);
```

Comment:
I would extract this to another method `getIdentifierFromImplementationClassName` or something similar.